### PR TITLE
feat: add shared error schema

### LINKED ISSUE #142: [feature] add option to share schema
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

As all errors follows the same format, it would be useful to have an option similar to what exists in @fastify/multipart to provide a shared schema for errors, for those of us who use schemas for validation/swagger.

For example, using the option:
```ts
{
  sharedSchemaId: '#httpErrors',
}
```

Would allow to reference this in all route schemas.

### Motivation

If you use schema for swagger, you'll have to write this schema in all your fastify apps.
This could be built-in in this plugin as it provides the error facilities. It would also make code most robust, in case you decide later to change the error response format, it would not break apps.

### Example

_No response_

### PR DESCRIPTION
**Pull request by sinedied**
Closes #142

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -53,6 +53,19 @@ function fastifySensible (fastify, opts, next) {
     }
   }
 
+  if (opts?.sharedSchemaId) {
+    fastify.addSchema({
+      $id: opts?.sharedSchemaId,
```
Comment (gurgunday) [inline index.js:58]: nit: `sharedSchemaId` is already defined inside the block

```suggestion
      $id: opts.sharedSchemaId,
```

Comment (sinedied) [reply index.js]: Fixed :)

**Review by gurgunday — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -53,6 +53,19 @@ function fastifySensible (fastify, opts, next) {
     }
   }
 
+  if (opts?.sharedSchemaId) {
+    fastify.addSchema({
```
Comment (Eomm) [inline index.js:57]:
```suggestion
    // The schema must be the same as:
    // https://github.com/fastify/fastify/blob/c08b67e0bfedc9935b51c787ae4cd6b250ad303c/build/build-error-serializer.js#L8-L16
    fastify.addSchema({
```

**Review by Eomm — APPROVED** [review]

**Pull request merged**